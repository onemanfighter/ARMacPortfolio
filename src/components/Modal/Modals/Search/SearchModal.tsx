import { Box, Text } from '@chakra-ui/react';

const SearchModal = () => {
  return (
    <Box>
      <Text>Search</Text>
    </Box>
  );
};

export default SearchModal;
