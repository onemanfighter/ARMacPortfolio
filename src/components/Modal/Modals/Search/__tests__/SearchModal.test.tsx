import { fireEvent, render, screen } from '@testing-library/react';
import SearchModal from '../SearchModal';

describe('SearchModal', () => {
  it('should render correctly', () => {
    const { container } = render(<SearchModal />);

    expect(container).toMatchSnapshot();
    expect(screen.getByLabelText('search-modal')).toBeDefined();
  });

  it('should not render component if isOpen is false', async () => {
    const { container } = render(<SearchModal />);

    fireEvent.change(screen.getByLabelText('search-input'), {
      target: { value: 'the' },
    });
    await jest.runAllTimersAsync();

    expect(container).toMatchSnapshot();
    expect(screen.getByLabelText('search-result')).toBeDefined();
  });
});
