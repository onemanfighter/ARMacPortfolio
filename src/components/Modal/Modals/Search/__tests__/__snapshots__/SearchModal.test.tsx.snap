// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchModal should not render component if isOpen is false 1`] = `
<div>
  <div
    aria-label="search-modal"
    class="css-1a4yqyo"
  >
    <div
      class="chakra-input__group css-bx0blc"
      data-group="true"
    >
      <div
        class="chakra-input__left-element css-1d6efr4"
      >
        <svg
          color="gray"
          fill="currentColor"
          height="1.5em"
          viewBox="0 0 24 24"
          width="1.5em"
        >
          <path
            d="M10 18a7.952 7.952 0 004.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0018 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
          />
        </svg>
      </div>
      <input
        aria-label="search-input"
        class="chakra-input css-139ybwu"
        placeholder="Search..."
        type="text"
        value="the"
      />
    </div>
    <div
      aria-label="search-result"
      class="css-0"
    >
      <div
        class="css-0"
      >
        The Lock Artist
      </div>
      <div
        class="css-0"
      >
        The Grand Design
      </div>
      <div
        class="css-0"
      >
        The Silmarillion
      </div>
      <div
        class="css-0"
      >
        The Preservationist
      </div>
      <div
        class="css-0"
      >
        The DaVinci Code
      </div>
      <div
        class="css-0"
      >
        The Lost Symbol
      </div>
      <div
        class="css-0"
      >
        The Book of Lies
      </div>
      <div
        class="css-0"
      >
        The Book of Samson
      </div>
      <div
        class="css-0"
      >
        The Code of the Wooster
      </div>
      <div
        class="css-0"
      >
        Lamb
      </div>
      <div
        class="css-0"
      >
        Fool
      </div>
      <div
        class="css-0"
      >
        Incompetence
      </div>
      <div
        class="css-0"
      >
        Monster 1959
      </div>
      <div
        class="css-0"
      >
        Thank You Jeeves
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchModal should render correctly 1`] = `
<div>
  <div
    aria-label="search-modal"
    class="css-1a4yqyo"
  >
    <div
      class="chakra-input__group css-bx0blc"
      data-group="true"
    >
      <div
        class="chakra-input__left-element css-1d6efr4"
      >
        <svg
          color="gray"
          fill="currentColor"
          height="1.5em"
          viewBox="0 0 24 24"
          width="1.5em"
        >
          <path
            d="M10 18a7.952 7.952 0 004.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0018 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
          />
        </svg>
      </div>
      <input
        aria-label="search-input"
        class="chakra-input css-139ybwu"
        placeholder="Search..."
        type="text"
        value=""
      />
    </div>
    <div
      aria-label="search-result"
      class="css-0"
    />
  </div>
</div>
`;
